packages/react-next/src/components/SwatchColorPicker/SwatchColorPicker.base.tsx

```diff
@@ -16,361 +8,341 @@ import {
+    focusedContainer: classNames.focusedContainer,
+  };
 
   /**
    * When the whole swatchColorPicker is blurred,
    * make sure to clear the pending focused stated
    */
-  private _onSwatchColorPickerBlur = (): void => {
-    if (this.props.onCellFocused) {
-      this._cellFocused = false;
-      this.props.onCellFocused();
+  const onSwatchColorPickerBlur = React.useCallback((): void => {
+    if (props.onCellFocused) {
+      internalState.cellFocused = false;
+      props.onCellFocused();
     }
-  };
-
-  /**
-   * Get the selected item's index
-   * @param items - The items to search
-   * @param selectedId - The selected item's id to find
-   * @returns - The index of the selected item's id, -1 if there was no match
-   */
-  private _getSelectedIndex(items: IColorCellProps[], selectedId: string): number | undefined {
-    const selectedIndex = findIndex(items, item => item.id === selectedId);
-    return selectedIndex >= 0 ? selectedIndex : undefined;
-  }
-
-  /**
-   * Render a color cell
-   * @param item - The item to render
-   * @returns - Element representing the item
-   */
-  private _renderOption = (item: IColorCellProps): JSX.Element => {
-    const props = this.props;
-    const id = this._id;
-
-    return (
-      <ColorPickerGridCell
-        item={item}
-        idPrefix={id}
-        color={item.color}
-        styles={props.getColorGridCellStyles}
-        disabled={props.disabled}
-        onClick={this._onCellClick}
-        onHover={this._onGridCellHovered}
-        onFocus={this._onGridCellFocused}
-        selected={this.state.selectedIndex !== undefined && this.state.selectedIndex === item.index}
-        circle={props.cellShape === 'circle'}
-        label={item.label}
-        onMouseEnter={this._onMouseEnter}
-        onMouseMove={this._onMouseMove}
-        onMouseLeave={this._onMouseLeave}
-        onWheel={this._onWheel}
-        onKeyDown={this._onKeyDown}
-        height={props.cellHeight}
-        width={props.cellWidth}
-        borderWidth={props.cellBorderWidth}
-      />
-    );
-  };
+  }, [props, internalState]);
 
   /**
    * Callback passed to the GridCell that will manage triggering the onCellHovered callback for mouseEnter
    */
-  private _onMouseEnter = (ev: React.MouseEvent<HTMLButtonElement>): boolean => {
-    if (!this.props.focusOnHover) {
-      return !this.isNavigationIdle || !!this.props.disabled;
-    }
-
-    if (this.isNavigationIdle && !this.props.disabled) {
-      ev.currentTarget.focus();
-    }
-
-    return true;
-  };
+  const onMouseEnter = React.useCallback(
+    (ev: React.MouseEvent<HTMLButtonElement>): boolean => {
+      if (!props.focusOnHover) {
+        return !internalState.isNavigationIdle || !!props.disabled;
+      }
+      if (internalState.isNavigationIdle && !props.disabled) {
+        ev.currentTarget.focus();
+      }
+      return true;
+    },
+    [props.focusOnHover, internalState.isNavigationIdle, props.disabled],
```

Comment:
Here you should just depend on `internalState` itself, not the navigation idle value. The final dep list once you fix the destructuring will be like this.
```suggestion
    [focusOnHover, internalState, disabled],
```